src/mlpack/methods/rbm/binary_layer.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file binary_rbm.hpp
+ * @author Kris Singh
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBM_BINARY_LAYER_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBM_BINARY_LAYER_HPP
+
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/base_layer.hpp>
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/activation_functions/logistic_function.hpp>
+
+using namespace mlpack;
+using namespace mlpack::ann;
+
+namespace mlpack{
+namespace rbm{
+
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class BinaryLayer
```

Comment:
I don't see a good reason for this. Also Layer is independent of the policy for eg a layer doesn't have access to the hidden layer in anyway. A new policy can have one layer as Binary and other spikeslab layer.